wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/GroupDetailsViewController.swift

```diff
@@ -59,9 +59,23 @@ final class GroupDetailsViewController: UIViewController, ZMConversationObserver
             }
 
             token = ConversationChangeInfo.add(observer: self, for: conversation)
+
             if let session = ZMUserSession.shared() {
-                syncObserver = InitialSyncObserver(in: session) { [weak self] completed in
-                    self?.didCompleteInitialSync = completed
+                if session.hasCompletedInitialSync {
+                    didCompleteInitialSync = true
+                } else {
+                    let context = session.managedObjectContext
+
+                    initialSyncToken = NotificationCenter.default.addObserver(
+                        forName: .initialSync,
+                        object: context.notificationContext,
+                        queue: nil
+                    ) { [weak self] _ in
+                        context.performGroupedBlock {
+                            self?.didCompleteInitialSync = true
+                            self?.initialSyncToken = nil
```

Comment:
Same here, `initialSyncToken` must be passed to `removeObserver`.
Setting the property to `nil` is not sufficient.